operate/qa/integration-tests/src/test/java/io/camunda/operate/elasticsearch/FinishedImportingIT.java

```diff
@@ -182,6 +172,47 @@ final var record = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 1);
         .until(() -> isRecordReaderIsCompleted("2-process-instance"));
   }
 
+  @Test
+  public void shouldNotSetCompletedToFalseForSubsequentRecordsAfterImportingDone()
+      throws IOException {
+    final var record = generateRecord(ValueType.PROCESS_INSTANCE, "8.6.0", 1);
+    EXPORTER.export(record);
+    esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+
+    zeebeImporter.performOneRoundOfImport();
+
+    // when
+    final var record2 = generateRecord(ValueType.PROCESS_INSTANCE, "8.7.0", 1);
+    EXPORTER.export(record2);
+    esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+
+    // receives 8.7 record and marks partition as finished importing
+    zeebeImporter.performOneRoundOfImport();
+
+    // then
+    // require multiple checks to avoid race condition. If records are written to zeebe indices and
+    // before a refresh, the record reader pulls the import batch is empty so it then says that the
+    // record reader is done when it is not.
+    for (int i = 0; i < RecordsReaderHolder.MINIMUM_EMPTY_BATCHES_FOR_COMPLETED_READER; i++) {
+      zeebeImporter.performOneRoundOfImport();
+    }
+
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(30))
+        .until(() -> isRecordReaderIsCompleted("1-process-instance"));
+
+    final var record3 = generateRecord(ValueType.PROCESS_INSTANCE, "8.7.0", 1);
+    EXPORTER.export(record3);
+    esClient.indices().refresh(new RefreshRequest("*"), RequestOptions.DEFAULT);
+
+    zeebeImporter.performOneRoundOfImport();
+
+    Awaitility.await()
+        .during(Duration.ofSeconds(10))
```

Comment:
🔧 Here we wait again for 10 seconds, I think we removed this the last time.

Reply:
It's not ideal but we really want to make sure this is not set to false and if we assert that it is true we could get a race condition. I have tried the following 
<img width="1043" alt="image" src="https://github.com/user-attachments/assets/7d7e636d-b701-4cc5-88af-812b380f5243">
where we fresh the indices and then check that it's true but this is a false positive where the test passes while the actual behaviour is wrong.

Reply:
An alternative I can think of is a thread.sleep but it's basically the same issue, this comes does to ES behaviour where an operation is not immediately executed and we must wait for the refresh timer.

Reply:
:D I had this discussion in my life so often now.

My learning so far: We can't assert something that shouldn't happen. You can assert something existing or a behavior you can observe. Either we rethink the test or leave it as it is.